metadata-ingestion/tests/unit/sdk_v2/test_lineage_client.py

```diff
@@ -561,3 +562,261 @@ def test_add_datajob_lineage_validation(client: DataHubClient) -> None:
         client.lineage.add_datajob_lineage(
             datajob=datajob_urn, downstreams=[invalid_urn]
         )
+
+
+def test_add_lineage_dataset_to_dataset_copy_basic(client: DataHubClient) -> None:
+    """Test add_lineage method with dataset to dataset and various column lineage strategies."""
+    upstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,upstream_table,PROD)"
+    downstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,downstream_table,PROD)"
+
+    # Test case 1: Basic dataset to dataset copy lineage (no column mapping)
+    client.lineage.add_lineage(upstream=upstream, downstream=downstream)
+    assert_client_golden(client, _GOLDEN_DIR / "test_lineage_copy_basic_golden.json")
+
+
+def test_add_lineage_dataset_to_dataset_copy_auto_fuzzy(client: DataHubClient) -> None:
+    upstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,upstream_table,PROD)"
+    downstream = "urn:li:dataset:(urn:li:dataPlatform:snowflake,downstream_table,PROD)"
+
+    upstream_schema = SchemaMetadataClass(
+        schemaName="upstream_table",
+        platform="urn:li:dataPlatform:snowflake",
+        version=1,
+        hash="1234567890",
+        platformSchema=OtherSchemaClass(rawSchema=""),
+        fields=[
+            SchemaFieldClass(
+                fieldPath="id",
+                type=SchemaFieldDataTypeClass(type=StringTypeClass()),
+                nativeDataType="string",
+            ),
+        ],
+    )
+
+    downstream_schema = SchemaMetadataClass(
+        schemaName="downstream_table",
+        platform="urn:li:dataPlatform:snowflake",
+        version=1,
+        hash="1234567890",
+        platformSchema=OtherSchemaClass(rawSchema=""),
+        fields=[
+            SchemaFieldClass(
+                fieldPath="id",
+                type=SchemaFieldDataTypeClass(type=StringTypeClass()),
+                nativeDataType="string",
+            ),
+        ],
+    )
+
+    # Test case 2: Auto fuzzy column lineage
```

Comment:
these numbers are going to get out of date quickly

imo the test names should be clear, concise, and self-explanatory